### PR TITLE
Better voice interface when there is one station option

### LINKED ISSUE #44: Better voice interface when there is one station option
When it's close to closing, there may only be 1 train arriving at the requested station. This is how the current voice interface works:

> When is the next train arriving at Ballston?
> Are you going to Vienna?
> Yes.
> Sorry, there was a problem with the requested skills response.

Maybe switch to:
> When is the next train arriving at Ballston?
> The next train from Ballston to Vienna arrives in 8 minutes. There are no other trains arriving at Ballston in the next 20 minutes.

### PR DESCRIPTION
**Pull request by pmyers88**
Fixes #44

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -30,3 +31,18 @@ exports.joinListConjuction = function (list, separator, conjuction) {
...
+  var responseText = '';
+  if (_.size(trainArrivals) === 0) {
+    responseText = props.stationNoArrivalsSpeechOutput;
+  } else if (_.size(trainArrivals) === 1) {
+    var destinationStationName = _.keys(trainArrivals)[0];
+    responseText = 'The next train to ' + destinationStationName + ' arrives in ' +
+      trainArrivals[destinationStationName] + ' minutes. There are no other trains arriving at ' + stationName +
+      ' in the next 20 minutes.';
+  } else {
+    responseText = 'Are you going to ' + this.joinListConjuction(_.keys(trainArrivals), ', ', ' or ');
```
Comment (r33beers) [inline lib/utils.js]: What does the trainArrivals object look like? Could you have two `Vienna` keys? If so, having a `size > 1` does not always mean you need to ask the user the direction.

Comment (r33beers) [inline lib/utils.js]: Never mind. I just saw the test case object.

Comment (r33beers) [inline lib/utils.js]: Add a `'?'` to the end of this message line. Maybe the Echo will use intonation.

File: test/test-utils.js [hunk 2]
```diff
@@ -43,3 +44,23 @@ test('joinListConjuction test', function (t) {
...
+  var oneStationArrival = {
+    'vienna': ['7']
+  };
+  var stationName = 'ballston';
+
+  t.equal(utils.makeGetStationResponseText(multipleStationArrivals, stationName), 'Are you going to largo town ' +
+    'center, vienna, wiehle reston east or new carrollton');
+  t.equal(utils.makeGetStationResponseText(oneStationArrival, stationName), 'The next train to vienna arrives in 7 ' +
+    'minutes. There are no other trains arriving at ballston in the next 20 minutes.');
+  t.equal(utils.makeGetStationResponseText({}, stationName), props.stationNoArrivalsSpeechOutput);
```
Comment (r33beers) [inline test/test-utils.js]: Let's hard code the expected response instead of loading from properties. That way, the test breaks if the property is changed. Then, we have to consciously choose to fix the test if it is the correct change.

File: test/test-utils.js [hunk 3]
```diff
@@ -43,3 +44,23 @@ test('joinListConjuction test', function (t) {
   t.equal(utils.joinListConjuction(things, ', ', ' and '), 'this, that and other');
   t.equal(utils.joinListConjuction(thing, ', ', ' and '), 'this');
 });
+
+test('makeGetStationResponseText tests', function (t) {
```
Comment (r33beers) [inline test/test-utils.js]: Can we get a bit more descriptive with the name of this test plan? Maybe `'test makeGetStationresponseText results for zero, single, or multiple stations'`

File: lib/metro-transit.js [hunk 4]
```diff
@@ -57,12 +57,14 @@ MetroTransit.prototype.intentHandlers = {
         }, {});
 
         console.info('Train Arrivals', trainArrivals);
-        if (_.size(trainArrivals) === 0) {
-          response.tell(props.stationNoArrivalsSpeechOutput);
+        var responseText = utils.makeGetStationResponseText(trainArrivals, stationName);
+        // if there's <= 1 train arriving we don't need any more info from the user; we can just tell them when that
+        // train is coming or that no trains are coming. Otherwise we're going to need to respond w/ a question.
+        if (_.size(trainArrivals) <= 1) {
```
Comment (r33beers) [inline lib/metro-transit.js]: nit: to me, it makes more sense to reverse this, so the condition is if > 1 ask for more info, else return because you know the answer.

File: lib/utils.js (hunk 1 above)
Comment (pmyers88) [inline lib/utils.js]: Yup it will, good call.

File: test/test-utils.js (hunk 2 above)
Comment (pmyers88) [inline test/test-utils.js]: Will do

File: package.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "description": "Amazon Echo skill for the Washington D.C. Metro",
   "main": "index.js",
   "scripts": {
-    "clean-deploy": "npm run clean && npm run test && npm run build && npm run deploy",
+    "clean-deploy": "npm run clean && npm run build && npm run deploy",
```
Comment (r33beers) [inline package.json]: Why did you decide to do this? I've always built something immediately after it passed tests.

Comment (pmyers88) [inline package.json]: Didn't mean to commit that.
